src/govuk/all.mjs

```diff
@@ -28,58 +28,76 @@ function initAll (config) {
 
   var $accordions = $scope.querySelectorAll('[data-module="govuk-accordion"]')
   nodeListForEach($accordions, function ($accordion) {
-    new Accordion($accordion, config.accordion).init()
+    if ($accordion instanceof HTMLElement) {
```

Comment:
Move these checks in to the individual component constructors where we currently have an existence check for `$module`.

Reply:
Brill, all moved and pushed up again

I've widened the `$accordion` (and other module) params to include `Element` (not just `HTMLElement`)

We can then narrow this down back to `HTMLElement` (as above) but inside the component constructors

**JavaScript style guide** updated too
[docs/contributing/coding-standards/js.md](https://github.com/alphagov/govuk-frontend/blob/fix-types/docs/contributing/coding-standards/js.md)